### DISCUSSION TITLE
Add support for confirmation prompt when sending multiple files

### DISCUSSION THREAD

**Discussion by Kabouik** [d99kris/nchat#161] [Ideas]
A warning when sending many (or just several) files at once using a file picker would be *very* convenient. I sent more than 550 images to a contact just because I had mistakenly pressed `a` in `nnn` which I use as an external file picker (`a` selects all files in the current folder), noticed it, and therefore pressed `q` to abort picking instead of deselecting the files… Mistake: `q` still picks the selected files in `nnn`. I alt-tabbed before the first file was upladed and left my phone there, so I didn't notice my mistake immediately, only to realize later that it had sent all those files while unattended. Even if I did notice it, there would have been no way to abort the upload (except maybe quitting `nchat`?).

No harm there, the contact was a trusted one and I could delete everything (it was faster from `telegram-desktop` thanks to multiselection), but the absence of a confirmatioin prompt could result in catastrophic data leakage in many cases. It would be very very cool if `nchat` could list all picked files before starting the multiple upload if ≥ 2 files are selected, with a confirmation prompt, maybe as an option.

I believe this would be a critical feature even if it only triggers when using an external picker and mistakenly manipulate it, given that such unconfirmed and massive leakage could only happen with `nchat` and not with `telegram-desktop` for instance.

**Comment by d99kris** [comment]
Hi @Kabouik - thanks for suggesting a feature. I have considered the suggestion, but I think I'll need to exclude it from nchat scope. The internal file-picker (with all its limitations) does not have the issue as it only allows single file selection, so it's limited to when using external file pickers. One can create a script or small app wrapping nnn/ranger/etc that prompts to confirm selections of more than X files, but I feel that is best implemented outside of nchat, since nnn/ranger/etc are already outside of nchat.
